packages/x-date-pickers/src/PickersShortcuts/PickersShortcuts.tsx

```diff
@@ -85,7 +85,13 @@ function PickersShortcuts<TValue>(props: PickersShortcutsProps<TValue>) {
         {
           maxHeight: VIEW_HEIGHT,
           maxWidth: 200,
-          overflow: 'auto',
+          '@media (pointer: fine)': {
+            scrollbarGutter: 'stable',
+            overflow: 'hidden',
+            '&:hover, &:focus-within': {
+              overflow: 'auto',
```

Comment:
All of the demos are sadly still not usable on mobile. 🙈 
https://deploy-preview-13972--material-ui-x.netlify.app/x/react-date-pickers/shortcuts/#know-which-shortcut-has-been-selected
https://deploy-preview-13972--material-ui-x.netlify.app/x/react-date-pickers/shortcuts/#customization

For reference: https://github.com/mui/mui-x/blob/1be2f4caab254861ce227b6ab398af07e6a45a6b/packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx#L64-L71